kv/kvnemesis: TestKVNemesisSingleNode failed

kv/kvnemesis.TestKVNemesisSingleNode [failed](https://tanzanite.cluster.engflow.com/invocations/default/5f3706aa-b386-4ef7-9c64-d08ed1548e7e?testReportRun=10&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZuZW1lc2lzOmt2bmVtZXNpc190ZXN0) on master @ [1898adc059e9b3026fd5a7ed88dc41cbb006abfc](https://github.com/cockroachdb/cockroach/commits/1898adc059e9b3026fd5a7ed88dc41cbb006abfc):

Fatal error:

```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1b19252]

```
Stack: 

```
goroutine 5460 [running]:
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).recover(0xc0058b5918?, {0x719b570, 0xc0026c31a0})
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:230 +0x65
panic({0x4a51220?, 0xa2f6b00?})
	GOROOT/src/runtime/panic.go:914 +0x21f
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).recover(0xc002830000?, {0x719b570, 0xc0027f57d0})
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:230 +0x65
panic({0x4a51220?, 0xa2f6b00?})
	GOROOT/src/runtime/panic.go:914 +0x21f
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).SendWithWriteBytes.func1()
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_send.go:109 +0x148
panic({0x4a51220?, 0xa2f6b00?})
	GOROOT/src/runtime/panic.go:920 +0x270
github.com/cockroachdb/cockroach/pkg/kv/kvserver/concurrency.(*lockTableGuardImpl).isSameTxn(...)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/concurrency/lock_table.go:804
github.com/cockroachdb/cockroach/pkg/kv/kvserver/concurrency.(*lockTableGuardImpl).IsKeyLockedByConflictingTxn(0xc0014f3040, {0xc005ff1d10, 0x14, 0x0?}, 0x1)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/concurrency/lock_table.go:749 +0x452
github.com/cockroachdb/cockroach/pkg/kv/kvserver/concurrency.(*Guard).IsKeyLockedByConflictingTxn(0xc0058b6010?, {0xc005ff1d10?, 0xc0032a3680?, 0xc003d24780?}, 0x0?)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/concurrency/concurrency_manager.go:809 +0x27
github.com/cockroachdb/cockroach/pkg/kv/kvserver/batcheval.(*requestBoundLockTableView).IsKeyLockedByConflictingTxn(0xc005ff1d10?, {0xc005ff1d10?, 0x1a3587b?, 0xc0058b60b0?})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/batcheval/lock.go:255 +0x23
github.com/cockroachdb/cockroach/pkg/storage.(*pebbleMVCCScanner).isKeyLockedByConflictingTxn(0xc0025ba000, {0x719b570, 0xc0027f5ad0}, {0xc005ff1d10?, 0x7feceda4018c?, 0x8?})
	github.com/cockroachdb/cockroach/pkg/storage/pebble_mvcc_scanner.go:1807 +0xa2
github.com/cockroachdb/cockroach/pkg/storage.(*pebbleMVCCScanner).add(0xc0025ba000, {0x719b570, 0xc0027f5ad0}, {0xc005ff1d10, 0x14, 0x30}, {0xc005ff1d10, 0x22, 0x30}, {0x7feceda4018c, ...})
	github.com/cockroachdb/cockroach/pkg/storage/pebble_mvcc_scanner.go:1236 +0xc5
github.com/cockroachdb/cockroach/pkg/storage.(*pebbleMVCCScanner).getOne(0xc0025ba000, {0x719b570, 0xc0027f5ad0})
	github.com/cockroachdb/cockroach/pkg/storage/pebble_mvcc_scanner.go:827 +0xca5
github.com/cockroachdb/cockroach/pkg/storage.(*pebbleMVCCScanner).scan(0xc0025ba000, {0x719b570, 0xc0027f5ad0})
	github.com/cockroachdb/cockroach/pkg/storage/pebble_mvcc_scanner.go:658 +0xbc
github.com/cockroachdb/cockroach/pkg/storage.mvccScanToBytes({_, _}, {_, _}, {_, _, _}, {0xc002b6edf8, 0x14, 0x18}, ...)
	github.com/cockroachdb/cockroach/pkg/storage/mvcc.go:4435 +0x350
github.com/cockroachdb/cockroach/pkg/storage.mvccScanToKvs({_, _}, {_, _}, {_, _, _}, {0xc002b6edf8, 0x14, 0x18}, ...)
	github.com/cockroachdb/cockroach/pkg/storage/mvcc.go:4488 +0xfe
github.com/cockroachdb/cockroach/pkg/storage.MVCCScan({_, _}, {_, _}, {_, _, _}, {0xc002b6edf8, 0x14, 0x18}, ...)
	github.com/cockroachdb/cockroach/pkg/storage/mvcc.go:4733 +0x316
github.com/cockroachdb/cockroach/pkg/kv/kvserver/batcheval.Scan({_, _}, {_, _}, {{0x7226ec0, 0xc004969940}, {{0x17ab1075b2bc81cd, 0x0}, 0x0, {0x17ab1075b2bcf08a, ...}, ...}, ...}, ...)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/batcheval/cmd_scan.go:92 +0x95f
github.com/cockroachdb/cockroach/pkg/kv/kvserver.evaluateCommand({_, _}, {_, _}, {_, _}, _, _, {{0x17ab1075b2bc81cd, 0x0}, ...}, ...)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_evaluate.go:488 +0x2ea
github.com/cockroachdb/cockroach/pkg/kv/kvserver.evaluateBatch({_, _}, {_, _}, {_, _}, {_, _}, _, 0xc00481e900, ...)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_evaluate.go:305 +0xba5
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).executeReadOnlyBatchWithServersideRefreshes(_, {_, _}, {_, _}, {_, _}, _, _, 0xc0058ba4e8, ...)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_read.go:473 +0x3db
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).executeReadOnlyBatch(0xc000185900, {0x719b570, 0xc0027f5ad0}, 0xc00481e900, 0xc004ca4a90)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_read.go:135 +0x785
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).executeBatchWithConcurrencyRetries(0xc000185900, {0x719b570, 0xc0027f5ad0}, 0xc00481e900, 0x56093d0)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_send.go:512 +0x3a3
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).SendWithWriteBytes(0xc000185900, {0x719b570?, 0xc0027f5950?}, 0xc00481e900)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_send.go:185 +0x6c5
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).SendWithWriteBytes(0xc003ba9500, {0x719b570?, 0xc0027f5890?}, 0xc00481e900)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_send.go:193 +0x85c
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Stores).SendWithWriteBytes(0x1?, {0x719b570, 0xc0027f5890}, 0xc00481e900)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/stores.go:202 +0x85
github.com/cockroachdb/cockroach/pkg/server.(*Node).batchInternal(0xc00504dc00, {0x719b570?, 0xc0027f5830?}, {0xc0022410e0?}, 0xc00481e900)
	github.com/cockroachdb/cockroach/pkg/server/node.go:1374 +0x545
github.com/cockroachdb/cockroach/pkg/server.(*Node).Batch(0xc00504dc00, {0x719b570, 0xc0027f57d0}, 0xc00481e900)
	github.com/cockroachdb/cockroach/pkg/server/node.go:1508 +0x2b5
github.com/cockroachdb/cockroach/pkg/rpc.makeInternalClientAdapter.func1({0x719b570?, 0xc0027f57d0?}, {0x5172720?, 0xc00481e900?})
	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:704 +0x45
github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.ServerInterceptor.func12({0x719b570, 0xc0027f57d0}, {0x5172720, 0xc00481e900}, 0xc00547fd80, 0xc003925c38)
	github.com/cockroachdb/cockroach/pkg/util/tracing/grpcinterceptor/grpc_interceptor.go:97 +0x4b9
github.com/cockroachdb/cockroach/pkg/rpc.makeInternalClientAdapter.chainUnaryServerInterceptors.bindUnaryServerInterceptorToHandler.func4({0x719b570?, 0xc0027f57d0?}, {0x5172720?, 0xc00481e900?})
	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:815 +0x37
github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func3({0x719b570, 0xc0027f57d0}, {0x5172720, 0xc00481e900}, 0xc0027f57d0?, 0xc00547fda0)
	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:169 +0x76
github.com/cockroachdb/cockroach/pkg/rpc.makeInternalClientAdapter.chainUnaryServerInterceptors.bindUnaryServerInterceptorToHandler.func4({0x719b570?, 0xc0027f57d0?}, {0x5172720?, 0xc00481e900?})
	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:815 +0x37
github.com/cockroachdb/cockroach/pkg/rpc.kvAuth.unaryInterceptor({0xc002830000?, {{0x10000004b5d58?}, {0x71d3550?, 0xc005ecac00?}}}, {0x719b570, 0xc0027f57d0}, {0x5172720, 0xc00481e900}, 0xc00547fd80, 0xc00547fdc0)
	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/auth.go:95 +0x1b3
github.com/cockroachdb/cockroach/pkg/rpc.makeInternalClientAdapter.chainUnaryServerInterceptors.bindUnaryServerInterceptorToHandler.func4({0x719b570?, 0xc0027f57d0?}, {0x5172720?, 0xc00481e900?})
	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:815 +0x37
github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func1.1({0x719b570?, 0xc0027f57d0?})
	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:136 +0x36
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunTaskWithErr(0xc00436b170, {0x719b570?, 0xc0027f57d0}, {0x0?, 0x1?}, 0xc0022aaa80)
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:336 +0xd3
github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func1({0x719b570?, 0xc0027f57d0?}, {0x5172720?, 0xc00481e900?}, 0x57c325?, 0x30?)
	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:134 +0x8f
github.com/cockroachdb/cockroach/pkg/rpc.makeInternalClientAdapter.chainUnaryServerInterceptors.bindUnaryServerInterceptorToHandler.func4({0x719b570?, 0xc0027f57d0?}, {0x5172720?, 0xc00481e900?})
	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:815 +0x37
github.com/cockroachdb/cockroach/pkg/rpc.makeInternalClientAdapter.func2({0x719b570?, 0xc0027f57d0?}, {0x0?, 0xc0022aac08?}, {0x5172720?, 0xc00481e900?}, {0x5047d40?, 0xc002921400}, 0xc004b258c0?, {0x0, ...})
	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:714 +0x51
github.com/cockroachdb/cockroach/pkg/rpc.NewContext.ClientInterceptor.func8({0x719b570, 0xc0027f57d0}, {0x5303acc, 0x21}, {0x5172720, 0xc00481e900}, {0x5047d40, 0xc002921400}, 0x147ae92?, 0xc0038dcdc0, ...)
	github.com/cockroachdb/cockroach/pkg/util/tracing/grpcinterceptor/grpc_interceptor.go:228 +0x47a
github.com/cockroachdb/cockroach/pkg/rpc.getChainUnaryInvoker.func1({0x719b570, 0xc0027f57d0}, {0x5303acc, 0x21}, {0x5172720, 0xc00481e900}, {0x5047d40, 0xc002921400}, 0x24?, {0x0, ...})
	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:899 +0x14e
github.com/cockroachdb/cockroach/pkg/rpc.makeInternalClientAdapter.func3({0x719b570, 0xc0026c31a0}, 0xc00481e7e0, {0x0, 0x0, 0x0})
	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:784 +0x32f
github.com/cockroachdb/cockroach/pkg/rpc.internalClientAdapter.Batch(...)
	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:907
github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*grpcTransport).sendBatch(0xc0021ce2a0, {0x719b570, 0xc0026c31a0}, 0xb7?, {0x717fe60, 0xc0044e5920?}, 0xc00481e7e0)
	github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/transport.go:209 +0x1bc
github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*grpcTransport).SendNext(0xc0021ce2a0, {0x719b570, 0xc0026c31a0}, 0x719b570?)
	github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/transport.go:187 +0x87
github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*DistSender).sendToReplicas(0xc0038cd700, {0x719b570, 0xc0026c31a0}, 0xc00479bb00?, {0xc0050ce280, 0xc0015e76c0, 0xc0015e7730, 0x0, 0x0}, 0x0)
	github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/dist_sender.go:2407 +0x1136
github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*DistSender).sendPartialBatch(0xc0038cd700, {0x719b570?, 0xc0026c31a0}, 0xc00479bb00, {{0xc003cc94b8, 0x14, 0x18}, {0xc002b6edf8, 0x14, 0x18}}, ...)
	github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/dist_sender.go:1908 +0x7b2
github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*DistSender).sendPartialBatchAsync.func1({0x719b570, 0xc0026c31a0})
	github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/dist_sender.go:1785 +0x147
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2()
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:484 +0x13a
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx in goroutine 5246
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:475 +0x415
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestKVNemesisSingleNode
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestKVNemesisSingleNode1929350409
    test_log_scope.go:81: use -show-logs to present logs inline
    kvnemesis_test.go:327: seed: 629054170974032836
    kvnemesis_test.go:234: kvnemesis logging to /var/lib/engflow/worker/work/3/exec/_tmp/8b60891bd2789237c716ace82dc4384b/kvnemesis1343475667
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>run=10</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #115694 kv/kvnemesis: TestKVNemesisSingleNode failed [A-kv-transactions C-bug C-test-failure O-robot P-1 T-kv branch-release-23.2]
</p>
</details>
/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestKVNemesisSingleNode.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-35391

kv/kvnemesis.TestKVNemesisSingleNode [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_UnitTests_BazelUnitTests/13575200?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_UnitTests_BazelUnitTests/13575200?buildTab=artifacts#/) on master @ [1898adc059e9b3026fd5a7ed88dc41cbb006abfc](https://github.com/cockroachdb/cockroach/commits/1898adc059e9b3026fd5a7ed88dc41cbb006abfc):


```
        	            	  | github.com/cockroachdb/cockroach/pkg/storage.mvccScanToKvs
        	            	  | 	github.com/cockroachdb/cockroach/pkg/storage/mvcc.go:4488
        	            	  | github.com/cockroachdb/cockroach/pkg/storage.MVCCScan
        	            	  | 	github.com/cockroachdb/cockroach/pkg/storage/mvcc.go:4733
        	            	  | github.com/cockroachdb/cockroach/pkg/kv/kvserver/batcheval.ReverseScan
        	            	  | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/batcheval/cmd_reverse_scan.go:89
        	            	  | github.com/cockroachdb/cockroach/pkg/kv/kvserver.evaluateCommand
        	            	  | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_evaluate.go:488
        	            	  | github.com/cockroachdb/cockroach/pkg/kv/kvserver.evaluateBatch
        	            	  | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_evaluate.go:305
        	            	  | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).executeReadOnlyBatchWithServersideRefreshes
        	            	  | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_read.go:473
        	            	  | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).executeReadOnlyBatch
        	            	  | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_read.go:135
        	            	  | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).executeBatchWithConcurrencyRetries
        	            	  | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_send.go:512
        	            	  | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).SendWithWriteBytes
        	            	  | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_send.go:185
        	            	  | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).SendWithWriteBytes
        	            	  | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_send.go:193
        	            	  | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Stores).SendWithWriteBytes
        	            	  | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/stores.go:202
        	            	  | github.com/cockroachdb/cockroach/pkg/server.(*Node).batchInternal
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/node.go:1374
        	            	  | github.com/cockroachdb/cockroach/pkg/server.(*Node).Batch
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/node.go:1508
        	            	  | github.com/cockroachdb/cockroach/pkg/rpc.makeInternalClientAdapter.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:704
        	            	  | github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.ServerInterceptor.func12
        	            	  | 	github.com/cockroachdb/cockroach/pkg/util/tracing/grpcinterceptor/grpc_interceptor.go:97
        	            	Wraps: (2) secondary error attachment
        	            	  | runtime error: invalid memory address or nil pointer dereference
        	            	  | (1) runtime error: invalid memory address or nil pointer dereference
        	            	  | Error types: (1) runtime.errorString
        	            	Wraps: (3) panic applying step 
        	            	  | db0.Txn(ctx, func(ctx context.Context, txn *kv.Txn) error {
        	            	  |   txn.SetIsoLevel(isolation.Snapshot)
        	            	  |   txn.GetForUpdate(ctx, tk(2954581848373496430)) // @1705475860.684367306,0 (<nil>, <nil>)
        	            	  |   txn.GetForShare(ctx, tk(574545114681662749)) // @1705475860.684367306,0 (v9, <nil>)
        	            	  |   txn.ReverseScanForShareSkipLocked(ctx, tk(1820498217220570453), tk(7831833396965981956), 0)
        	            	  |   txn.ScanForUpdate(ctx, tk(10391090840890742608), tk(14359564960357966375), 0)
        	            	  |   txn.ReverseScanForUpdate(ctx, tk(7224580107135627516), tk(13038096599029329166), 0)
        	            	  |   txn.GetForShareGuaranteedDurability(ctx, tk(12530610722970564363))
        	            	  |   txn.ScanForShare(ctx, tk(7262151861797917607), tk(18430215113825785752), 0)
        	            	  |   return errors.New("rollback")
        	            	  | }): runtime error: invalid memory address or nil pointer dereference
        	            	Error types: (1) *withstack.withStack (2) *secondary.withSecondaryError (3) *errutil.leafError
    panic.go:523: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/1f42cf5be2fc021646bf9b2daf5eaef3/logTestKVNemesisSingleNode3703101735
--- FAIL: TestKVNemesisSingleNode (3.34s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #115694 kv/kvnemesis: TestKVNemesisSingleNode failed [A-kv-transactions C-bug C-test-failure O-robot P-1 T-kv branch-release-23.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestKVNemesisSingleNode.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/kv/kvserver/concurrency/lock_table.go
+++ b/pkg/kv/kvserver/concurrency/lock_table.go
@@ -746,7 +746,7 @@ func (g *lockTableGuardImpl) IsKeyLockedByConflictingTxn(
 			// queuedLockingRequests is sorted in increasing order of sequence number.
 			break
 		}
-		if g.isSameTxn(qqg.guard.txnMeta()) {
+		if qqg.guard.txnMeta() != nil && g.isSameTxn(qqg.guard.txnMeta()) {
 			// A SKIP LOCKED request should not find another waiting request from its
 			// own transaction, at least not in the way that SQL uses KV. The only way
 			// we can end up finding another request in the lock's wait queue from our

